modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableColumnHeader.java

```diff
@@ -581,13 +585,187 @@ private void initUI() {
         }
     }
 
-    private void doColumnAutoSize(TableColumnBase<?,?> column, int cellsToMeasure) {
-        double prefWidth = column.getPrefWidth();
+    private void doColumnAutoSize(int cellsToMeasure) {
+        double prefWidth = getTableColumn().getPrefWidth();
 
         // if the prefWidth has been set, we do _not_ autosize columns
         if (prefWidth == DEFAULT_COLUMN_WIDTH) {
-            TableSkinUtils.resizeColumnToFitContent(getTableSkin(), column, cellsToMeasure);
-//            getTableViewSkin().resizeColumnToFitContent(column, cellsToMeasure);
+            resizeColumnToFitContent(cellsToMeasure);
+        }
+    }
+
+    /**
+     * Resizes this {@code TableColumnHeader}'s column based on content of header and content of cells. This
+     * implementation measures the preferred width of the header, the preferred width of the first {@code maxRow} cells
+     * and sizes the column to the maximum width of all measured values. Subclass can either call this method or
+     * override it (no need to call {@code super()}) to provide their custom implementation (exclude headers, exclude
+     * null content, use min).
+     *
+     * @param maxRows the number of rows considered when resizing. If -1 is given, all rows are considered.
+     * @since 14
```

Comment:
I have put the `since 14`, because if merged, it will be available in OpenJFX 14 right? (It was 13 before)

Reply:
Correct.

Reply:
The "The resulting column width for this implementation..." part might belong in an `@implSpec` section.

Reply:
From https://openjdk.java.net/jeps/8068562

> Implementation Specification. This is where the default implementation (or an overrideable implementation in a class) is specified. Interface implementors or class subclassers use the information here in order to decide whether it is sensible or necessary to override a particular method, and what behavior they can rely on if a method is called via super.

So maybe the whole part beginning from "The resulting column [..]width etc.)." should be in the `@implSpec` section.

Reply:
Yes.